src/main/java/in/org/projecteka/hiu/dataflow/DataFlowRequestListener.java

```diff
@@ -61,14 +68,26 @@ public void subscribe() {
 
                 logger.info("Initiating data flow request to consent manager");
                 centralRegistry.token()
-                        .flatMap(token -> dataFlowClient.initiateDataFlowRequest(dataFlowRequest, token)
-                                .flatMap(dataFlowRequestResponse ->
-                                        dataFlowRepository.addDataRequest(dataFlowRequestResponse.getTransactionId(),
-                                                consentId,
-                                                dataFlowRequest)
-                                                .then(dataFlowRepository.addKeys(
-                                                        dataFlowRequestResponse.getTransactionId(),
-                                                        dataRequestKeyMaterial)))).block();
+                        .flatMap(token -> {
+                            if (dataFlowProperties.isUsingGateway()) {
+                                var gatewayDataFlowRequest = getDataFlowRequest(dataFlowRequest);
+                                return consentRepository.getPatientId(consentId)
+                                        .flatMap(patientId -> dataFlowClient.initiateDataFlowRequest(gatewayDataFlowRequest, token, getCmSuffix(patientId)))
+                                        .then(Mono.defer(() -> {
+                                            dataFlowCache.put(gatewayDataFlowRequest.getRequestId().toString(), dataFlowRequest);
```

Comment:
Modified the code to store key material along with data flow request. Those changes will come with our PR.
You don't need to update it :)